src/main/groovy/life/qbic/samplestatus/reporter/SampleUpdate.groovy

```
@@ -0,0 +1,22 @@
+package life.qbic.samplestatus.reporter
+
+import groovy.transform.EqualsAndHashCode
+
+import java.time.Instant
+
+/**
+ * <b>Value Class SampleUpdate</b>
+ *
+ * @since 0.1.0
+ */
+@EqualsAndHashCode
+class SampleUpdate {
+    Sample sample
+    String updatedStatus
```

Question: Do you think it would make sense to change this from String to a dedicated Enum holding all possible values. (I guess it's dependent on how the statuses are stored in the LIMS)

---

No by doing that we would depend on data-model-lib again and would need adjustment anytime something changes. I think it is better to let the request to the service fail for unknown statuses and validate the status there.